pennylane/gradients/hamiltonian_grad.py

```diff
@@ -25,18 +25,35 @@ def hamiltonian_grad(tape, idx, params=None):
         idx (int): index of parameter that we differentiate with respect to
         params (array): explicit parameters to set
     """
+    op, p_idx = tape.get_operation(idx)
+    new_tape = tape.copy(copy_operations=True)
 
-    t_idx = list(tape.trainable_params)[idx]
-    op = tape._par_info[t_idx]["op"]
-    p_idx = tape._par_info[t_idx]["p_idx"]
-
-    new_tape = tape.copy(copy_operations=True, tape_cls=qml.tape.QuantumTape)
     if params is not None:
         new_tape.set_parameters(params=params)
 
-    new_tape._measurements = [qml.expval(op.ops[p_idx])]
+    # get position in queue
+
+    for i, m in enumerate(tape.measurements):
```

Comment:
Is there a way to do this using `index` or something, to avoid the `for` loop?

Reply:
oh good idea 🤔 

Considering that the list of measurements is probably O(1) - O(10), and `list.index` has [linear time complexity](https://stackoverflow.com/questions/5913671/complexity-of-list-indexx-in-python) I would expect both the `for`-`break` and `index` to take about the same amount of time, but indexing seems faster:

![image](https://user-images.githubusercontent.com/2959003/132280847-c27f9ddc-12d4-4430-a2ae-11da65d2b7a6.png)

Reply:
I've made this change :)